Sources/Datadog/Core/Persistence/DataBlock.swift

```diff
@@ -0,0 +1,193 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Block size binary type
+internal typealias BlockSize = UInt32
+
+/// Block type supported in data stream
+internal enum BlockType: UInt16 {
+    case event = 0x00
+    case headerV1 = 0x01
+}
+
+/// Reported errors while manipulateing data blocks.
+internal enum DataBlockError: Error {
+    case readOperationFailed(streamError: Error?)
+    case invalidByteSequence
+}
+
+/// A data block in defined by its type and a byte sequence.
+///
+/// A block can be serialized in data stream by following TLV format.
+internal struct DataBlock {
+    /// Type describing the data block.
+    let type: BlockType
+
+    /// The data.
+    var data: Data
+
+    /// Returns a Data block in Type-Lenght-Value format.
+    ///
+    /// A block follow TLV with bytes aligned such as:
+    ///
+    ///     +-  2 bytes -+-   4 bytes   -+- n bytes -|
+    ///     | block type | data size (n) |    data   |
+    ///     +------------+---------------+-----------+
+    ///
+    /// - Returns: a data block in TLV.
+    func serialize() -> Data {
+        var buffer = Data()
+        // T
+        withUnsafeBytes(of: type.rawValue) { buffer.append(contentsOf: $0) }
+        // L
+        withUnsafeBytes(of: BlockSize(data.count)) { buffer.append(contentsOf: $0) }
+        // V
+        buffer += data
+        return buffer
+    }
+}
+
+internal extension Data {
+    /// Returns a Data block in Type-Lenght-Value format.
+    ///
+    /// A block follow TLV with bytes aligned such as:
+    ///
+    ///     +-  2 bytes -+-   4 bytes    -+- n bytes  -|
+    ///     | block type | block size (n) | block data |
+    ///     +------------+----------------+------------+
+    ///
+    /// - Parameters:
+    ///   - type: The data type
+    ///   - data: The data
+    /// - Returns: a byte sequence in TLV format.
+    static func block(_ type: BlockType, data: Data) -> Data {
+        return DataBlock(type: type, data: data).serialize()
+    }
+}
```

Comment:
Is this extension needed 💭🤔? I can only see its usage in tests, not in the production code. Also, as we're using Foundation type (`Data`) to only attach a domain-specific (core) function to, so it doesn't seem to bring much improvement over using `DataBlock(type:data:)` directly, WDYT?

Reply:
Yea, not really needed, it was more for convenience for future change. I will remove it!
Could be cool to leverage a pattern like [AlamofireExtended](https://github.com/Alamofire/Alamofire/blob/master/Source/AlamofireExtended.swift) in the future, so we can hide any extension behind a `dd` property!